app/src/main/java/com/xmartlabs/template/ui/mvp/MvpPresenter.java

```diff
@@ -0,0 +1,24 @@
+package com.xmartlabs.template.ui.mvp;
+
+import android.support.annotation.UiThread;
+
+/**
+ * The base interface that defines a presenter in the MVP patterns
+ * @param <V> The view type to be attached to this presenter
+ */
+public interface MvpPresenter<V extends MvpView> {
```

Comment:
@diegomedina248 what about if we have an app which needs a more complex lifecycle behaviour , Shoudn't you create another interface which extends from `MvpPresenter` and which has more lifecycles events, like the fragment's lifecycle events.